### PR TITLE
[genio] Correct dnssd api usage

### LINKED ISSUE #23847: [genio] correct dnssd api usage
### Reproduction steps

```bash
1. Commission with multiple controller
2. Dnssd publish service can not publish service due to out of socket
```

### Platform

other

### Platform Version(s)

_No response_

### Type

Manually tested with SDK

### (Optional) If manually tested please explain why this is only manually tested

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by j8xixo12**
`Fixes #23847`
This PR contains following changes:
1. Correct dnssd api usage
2. remove unused variables

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
@j8xixo12 - needs restyle fixes.

File: src/platform/mt793x/DnssdImpl.h [hunk 1]
```diff
@@ -91,6 +90,23 @@ class MdnsContexts
...
 
+struct RegisterContext : public GenericContext
+{
+    DnssdPublishCallback callback;
+    std::string mType;
+    std::string mInstanceName;
+    // HostNameRegistrar mHostNameRegistrar;
+
+    RegisterContext(const char * sType, const char * instanceName, DnssdPublishCallback cb, void * cbContext);
+    // virtual ~RegisterContext() { mHostNameRegistrar.Unregister(); }
```
Comment (bzbarsky-apple) [inline src/platform/mt793x/DnssdImpl.h:101]: Why is this commented-out code here?

File: src/platform/mt793x/DnssdImpl.cpp [hunk 2]
```diff
@@ -55,25 +188,56 @@ namespace Dnssd {
...
+    {
+        auto err = DNSServiceUpdateRecord(sdCtx->serviceRef, nullptr, kRegisterFlags, record.size(), record.data(), 0 /* ttl */);
+        VerifyOrReturnError(kDNSServiceErr_NoError == err, sdCtx->Finalize(err));
+        return CHIP_NO_ERROR;
+    }
+
+    sdCtx = chip::Platform::New<RegisterContext>(type, name, callback, context);
+    VerifyOrReturnError(nullptr != sdCtx, CHIP_ERROR_NO_MEMORY);
+
+    // sdCtx->mHostNameRegistrar.Init(hostname, addressType, interfaceId);
```
Comment (bzbarsky-apple) [inline src/platform/mt793x/DnssdImpl.cpp:226]: Why is this commented-out code here?

File: src/platform/mt793x/DnssdImpl.cpp [hunk 3]
```diff
@@ -55,25 +188,56 @@ namespace Dnssd {
...
+
+    DNSServiceRef sdRef;
+    auto err = DNSServiceRegister(&sdRef, kRegisterFlags, interfaceId, name, type, kLocalDot, hostname, ntohs(port), record.size(),
```
Comment (bzbarsky-apple) [inline src/platform/mt793x/DnssdImpl.cpp:229]: Shouldn't that be `htons(port)`?

**Review by bzbarsky-apple — APPROVED** [review]
Mostly rubber-stamping.

File: src/platform/mt793x/DnssdImpl.h (hunk 1 above)
Comment (j8xixo12) [reply src/platform/mt793x/DnssdImpl.h]: Oh..., there are some messy code during development, I just removed it.

File: src/platform/mt793x/DnssdImpl.cpp (hunk 2 above)
Comment (j8xixo12) [reply src/platform/mt793x/DnssdImpl.cpp]: ditto.

File: src/platform/mt793x/DnssdImpl.cpp (hunk 3 above)
Comment (j8xixo12) [reply src/platform/mt793x/DnssdImpl.cpp]: Yes `htons(port)` is right!
I think I can passed PICS test due to my test devices are all **litte-endian**, however if test device contains **big-endian**, I guess PICS test will fail.

**Review by yufengwangca — APPROVED** [review]

**Pull request merged**